### ISSUE TITLE
Docker image with lychee executable

### ISSUE THREAD

**Issue by martinberoiz** [lycheeverse/lychee#1378]
Hello, apologies if this has been answered before.
I'm trying to set up a CI test on a GitLab site and I need a docker image that contains the `lychee` command.

I thought that the official [one in docker hub](https://hub.docker.com/r/lycheeverse/lychee) would do, but I realized that it's meant to be run as a containerized command line program. That's not exactly what I want.

Is there any bare (hopefully small) docker container with the `lychee` executable in it? Simliary to what the `python` image is.

I love the project and appreciate all the effort put on this!

**Comment by mre** [comment]
Hey @martinberoiz,

maybe the [alpine image](https://hub.docker.com/layers/lycheeverse/lychee/latest-alpine/images/sha256-4a028ef6271d9357233dd23650fed4c9dd691f91ff9d98b376a25c836f42e5c5?context=explore) would work? 

```sh
docker run lycheeverse/lychee:latest-alpine
```

It's 13MB in size and contains nothing but lychee.

If not, I'm not quite sure whether I understand the difference between a containerized command line program and a Docker image that contains the lychee command. Essentially, lychee is a command line program, dockerized or not. 😉

**Comment by martinberoiz** [comment]
Thanks for your quick reply. My work day is about to end, I'll try that tomorrow and come back to you. It could work.

**Comment by martinberoiz** [comment]
Ok, I tried and could not make it work. Specifically I cannot run it as a background process:

```shell
$ docker pull lycheeverse/lychee:latest-alpine
$ docker run -itd --name lychee lycheeverse/lychee:latest-alpine
$ docker exec lychee /bin/sh
Error response from daemon: Container 225fa7a29aaaf9be968fed89651f82db28a8f6508671b3d8eefbe2a18fcd167c is not running
```

I don't know enough of docker to understand what's going on, but I did my own [Dockerfile](https://github.com/martinberoiz/lychee-ci-docker/blob/master/Dockerfile) that satisfies my needs for the moment.

Please feel free to close this issue, unless you think it's worth keep digging. If I learn more and find a solution I'll try to open a PR or a new issue.

**Comment by martinberoiz** [comment]
[This SO](https://stackoverflow.com/questions/76263404/how-to-keep-the-entrypoint-but-replace-cmd-of-a-docker-image-from-a-gitlab-pipel) seems relevant.

tl;dr Gitlab CI will not work with images with an ENTRYPOINT. There is a workaround suggested there that I will try.

**Comment by martinberoiz** [comment]
For the record, this is the solution how to use it on `gitlab-ci.yaml`:

```
job-name:
  stage: my-stage
  image:
    name: lycheeverse/lychee:latest-alpine
    entrypoint: [""]
```

Sorry for the many replies!
